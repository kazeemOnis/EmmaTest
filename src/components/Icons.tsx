import React from 'react';
import Svg, {SvgProps, Path, G, Defs, ClipPath} from 'react-native-svg';
import {colors} from '../styles';

export const User = (props: SvgProps) => (
  <Svg width={20} height={20} fill="none" {...props}>
    <G fill={props.fill ? props.fill : colors.aqua} clipPath="url(#a)">
      <Path d="M10 10c1.38 0 2.559-.488 3.535-1.465C14.512 7.56 15 6.38 15 5c0-1.38-.488-2.559-1.465-3.535C12.56.488 11.38 0 10 0 8.62 0 7.441.488 6.465 1.465 5.488 2.44 5 3.62 5 5c0 1.38.488 2.559 1.465 3.535C7.44 9.512 8.62 10 10 10Z" />
      <Path d="M19.121 15.28c-.03-.438-.091-.911-.182-1.42a11.053 11.053 0 0 0-.345-1.412 6.705 6.705 0 0 0-.56-1.27 4.574 4.574 0 0 0-.808-1.054 3.399 3.399 0 0 0-1.113-.697 3.914 3.914 0 0 0-1.452-.26c-.078 0-.26.093-.547.28-.286.186-.61.395-.97.625-.36.23-.829.438-1.406.625a5.62 5.62 0 0 1-1.738.28 5.618 5.618 0 0 1-1.739-.28c-.577-.187-1.046-.395-1.406-.625-.36-.23-.683-.439-.97-.625-.286-.187-.469-.28-.547-.28-.53 0-1.013.087-1.452.26-.438.174-.809.406-1.113.697-.304.29-.573.642-.807 1.054-.234.413-.421.836-.56 1.27-.139.434-.254.905-.345 1.413-.091.508-.152.98-.182 1.42-.03.438-.046.887-.046 1.347 0 1.041.317 1.864.95 2.468.634.603 1.477.904 2.527.904h11.38c1.05 0 1.893-.301 2.526-.904.634-.604.951-1.426.951-2.468 0-.46-.015-.91-.046-1.348Z" />
    </G>
    <Defs>
      <ClipPath id="a">
        <Path fill="#fff" d="M0 0h20v20H0z" />
      </ClipPath>
    </Defs>
  </Svg>
);

export const PiggyBank = (props: SvgProps) => (
  <Svg width={30} height={30} fill="none" {...props}>
    <Path
      fill={props.fill ? props.fill : colors.pink}
      d="M22.903 12.69h-.859a6.61 6.61 0 0 0-2.588-3.22l.753-2.58-3.761 1.437a8.68 8.68 0 0 0-1.357-.109c-2.596 0-4.748 1.11-6.04 2.813a.585.585 0 0 0-.155-.024c-1.364 0-1.827-.936-1.851-.987a.584.584 0 0 0-1.067.476c.027.063.645 1.41 2.394 1.643a6.448 6.448 0 0 0-.636 2.804c0 2.123 1.007 4.015 2.667 5.248l-.414.413a1.468 1.468 0 1 0 2.076 2.076l1.206-1.205c.598.13 1.208.194 1.82.193a8.428 8.428 0 0 0 2.243-.296l1.308 1.308a1.468 1.468 0 0 0 2.076-2.076l-.645-.645a6.542 6.542 0 0 0 1.97-2.762h.86a1.169 1.169 0 0 0 1.17-1.17v-2.169a1.169 1.169 0 0 0-1.17-1.168Zm-4.172 1.393a.812.812 0 1 1 0-1.624.812.812 0 0 1 0 1.624Z"
    />
  </Svg>
);

export const Building = (props: SvgProps) => (
  <Svg width={20} height={20} fill="none" {...props}>
    <Path
      fill={props.fill ? props.fill : colors.aqua}
      d="m18.438 5.688-8.125-4.375c-.188-.125-.438-.125-.563 0L1.625 5.688c-.25.125-.375.312-.375.562v1.875c0 .375.25.625.625.625h16.25c.375 0 .625-.25.625-.625V6.25a.656.656 0 0 0-.313-.563ZM10.625 15c.375 0 .625-.25.625-.625V10c0-.375-.25-.625-.625-.625h-1.25c-.375 0-.625.25-.625.625v4.375c0 .375.25.625.625.625h1.25ZM15.625 15c.375 0 .625-.25.625-.625V10c0-.375-.25-.625-.625-.625h-1.25c-.375 0-.625.25-.625.625v4.375c0 .375.25.625.625.625h1.25ZM5.625 15c.375 0 .625-.25.625-.625V10c0-.375-.25-.625-.625-.625h-1.25c-.375 0-.625.25-.625.625v4.375c0 .375.25.625.625.625h1.25ZM18.125 15.625H1.875c-.375 0-.625.25-.625.625v1.875c0 .375.25.625.625.625h16.25c.375 0 .625-.25.625-.625V16.25c0-.375-.25-.625-.625-.625Z"
    />
  </Svg>
);

export const ChevronLeft = (props: SvgProps) => (
  <Svg width={10} height={16} fill="none" {...props}>
    <Path
      d="M9.016 13.734a1.083 1.083 0 1 1-1.532 1.532l-6.5-6.5a1.083 1.083 0 0 1-.033-1.498L6.91.768a1.083 1.083 0 0 1 1.597 1.464L3.25 7.967l5.766 5.767Z"
      fill={props.fill ? props.fill : colors.white}
    />
  </Svg>
);
