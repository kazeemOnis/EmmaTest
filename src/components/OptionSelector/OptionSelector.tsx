import React from 'react';
import {Dimensions} from 'react-native';
import styled from '../../theme';
import {colors} from '../../styles';

interface OptionProps {
  text: string;
  onPress: Function;
  icon?: JSX.Element;
}

const OptionWrapper = styled.TouchableOpacity`
  background-color: ${colors.secondary};
  padding: 12px;
  min-height: 140px;
  border-radius: 12px;
  width: ${Dimensions.get('window').width / 2 - 24}px;
`;

const Text = styled.Text`
  font-family: 'Nunito';
  color: white;
  font-size: 18px;
  width: 60%;
`;

const IconWrapper = styled.View`
  background-color: ${colors.primary};
  width: 40px;
  height: 40px;
  border-radius: 16px;
  justify-content: center;
  align-items: center;
  margin-bottom: 16px;
`;

export const OptionSelector = ({text, icon, onPress}: OptionProps) => {
  return (
    <OptionWrapper testID="optionSelector" onPress={() => onPress()}>
      {icon && <IconWrapper>{icon}</IconWrapper>}
      <Text numberOfLines={2}>{text}</Text>
    </OptionWrapper>
  );
};
