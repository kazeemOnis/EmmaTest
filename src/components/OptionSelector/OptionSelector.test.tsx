import React from 'react';
import {cleanup, render, fireEvent} from '@testing-library/react-native';
import {act} from 'react-test-renderer';
import {OptionSelector} from './OptionSelector';
import {User} from '../Icons';

afterEach(() => {
  cleanup();
});

const onPressMock = jest.fn();

const mockProps = {
  text: 'Contacts',
  onPress: onPressMock,
};

const setup = (overrides = {}) => {
  const props = {...mockProps, ...overrides};
  const utils = render(<OptionSelector {...props} />);
  return {...utils, props};
};

describe('OptionSelector', () => {
  it('Should render content', async () => {
    const {toJSON} = setup();
    await act(async () => {
      expect(toJSON()).toMatchSnapshot();
    });
  });

  it('Should render all text', () => {
    const {queryByText} = setup();

    expect(queryByText('Contacts')).toBeTruthy();
  });

  it('Should render an icon when provided', async () => {
    const {toJSON} = setup({
      icon: <User />,
    });
    await act(async () => {
      expect(toJSON()).toMatchSnapshot();
    });
  });

  it('Should click the option selector when pressed', async () => {
    const {getByTestId} = setup();
    fireEvent.press(getByTestId('optionSelector'));

    expect(onPressMock).toHaveBeenCalledTimes(1);
  });
});
