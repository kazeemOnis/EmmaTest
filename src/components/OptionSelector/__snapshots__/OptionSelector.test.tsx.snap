// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OptionSelector Should render an icon when provided 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#221D34",
      "borderRadius": 12,
      "minHeight": 140,
      "opacity": 1,
      "paddingBottom": 12,
      "paddingLeft": 12,
      "paddingRight": 12,
      "paddingTop": 12,
      "width": 351,
    }
  }
  testID="optionSelector"
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#16102A",
          "borderRadius": 16,
          "height": 40,
          "justifyContent": "center",
          "marginBottom": 16,
          "width": 40,
        },
      ]
    }
  >
    <RNSVGSvgView
      bbHeight="20"
      bbWidth="20"
      fill="none"
      focusable={false}
      height={20}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 20,
            "width": 20,
          },
        ]
      }
      width={20}
    >
      <RNSVGGroup
        fill={null}
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGGroup
          clipPath="a"
          fill={
            {
              "payload": 4284273613,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="M10 10c1.38 0 2.559-.488 3.535-1.465C14.512 7.56 15 6.38 15 5c0-1.38-.488-2.559-1.465-3.535C12.56.488 11.38 0 10 0 8.62 0 7.441.488 6.465 1.465 5.488 2.44 5 3.62 5 5c0 1.38.488 2.559 1.465 3.535C7.44 9.512 8.62 10 10 10Z"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
          />
          <RNSVGPath
            d="M19.121 15.28c-.03-.438-.091-.911-.182-1.42a11.053 11.053 0 0 0-.345-1.412 6.705 6.705 0 0 0-.56-1.27 4.574 4.574 0 0 0-.808-1.054 3.399 3.399 0 0 0-1.113-.697 3.914 3.914 0 0 0-1.452-.26c-.078 0-.26.093-.547.28-.286.186-.61.395-.97.625-.36.23-.829.438-1.406.625a5.62 5.62 0 0 1-1.738.28 5.618 5.618 0 0 1-1.739-.28c-.577-.187-1.046-.395-1.406-.625-.36-.23-.683-.439-.97-.625-.286-.187-.469-.28-.547-.28-.53 0-1.013.087-1.452.26-.438.174-.809.406-1.113.697-.304.29-.573.642-.807 1.054-.234.413-.421.836-.56 1.27-.139.434-.254.905-.345 1.413-.091.508-.152.98-.182 1.42-.03.438-.046.887-.046 1.347 0 1.041.317 1.864.95 2.468.634.603 1.477.904 2.527.904h11.38c1.05 0 1.893-.301 2.526-.904.634-.604.951-1.426.951-2.468 0-.46-.015-.91-.046-1.348Z"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
          />
        </RNSVGGroup>
        <RNSVGDefs>
          <RNSVGClipPath
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            name="a"
          >
            <RNSVGPath
              d="M0 0h20v20H0z"
              fill={
                {
                  "payload": 4294967295,
                  "type": 0,
                }
              }
              propList={
                [
                  "fill",
                ]
              }
            />
          </RNSVGClipPath>
        </RNSVGDefs>
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <Text
    numberOfLines={2}
    style={
      [
        {
          "color": "white",
          "fontFamily": "Nunito",
          "fontSize": 18,
          "width": "60%",
        },
      ]
    }
  >
    Contacts
  </Text>
</View>
`;

exports[`OptionSelector Should render content 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#221D34",
      "borderRadius": 12,
      "minHeight": 140,
      "opacity": 1,
      "paddingBottom": 12,
      "paddingLeft": 12,
      "paddingRight": 12,
      "paddingTop": 12,
      "width": 351,
    }
  }
  testID="optionSelector"
>
  <Text
    numberOfLines={2}
    style={
      [
        {
          "color": "white",
          "fontFamily": "Nunito",
          "fontSize": 18,
          "width": "60%",
        },
      ]
    }
  >
    Contacts
  </Text>
</View>
`;
